node-lib/src/options.rs

```diff
@@ -176,6 +178,48 @@ pub struct RunOptions {
     pub rpc_cookie_file: Option<String>,
 }
 
+impl RunOptions {
```

Comment:
Two things:

1. The resolution should be part of RPC, not part of RunOptions. So this should be done in the `rpc` crate.
2. It seems that the resolution process is exactly the same for both http and websocket. There should be a way to abstract both calls into one.

Though I want to add that I'm planning to remove websocket at some point because it's kind of pointless. We decided to go with polling instead of pub/sub for the sake of simplicity, which is the way the wallet currently works. So feel free to eliminate websocket as well to simplify this task too.